src/main/java/duke/DateTimeParser.java

```diff
@@ -0,0 +1,127 @@
+package duke;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoField;
+
+/**
+ * Contains functionality for parsing date and time input, as well as formatting
+ * LocalDate and LocalDateTime objects.
+ */
+public class DateTimeParser {
```

Comment:
I like your idea of creating a new class for parsing LocalDateTime. Maybe it will be better if the parser and datetimeparser are under the same packet?